.github/workflows/deliver-ubuntu.yml

```diff
@@ -0,0 +1,110 @@
+
+          - name: Metadata
+            id: metadata
+            run: |
+                git_description=$(git describe --tags --long)
+                version=$(echo "${git_description}" | awk -F- '{print $(1)}' | sed 's/^v//')
+                revision=$(echo "${git_description}" | awk -F- '{print $(NF-1)}')
+                commit=$(echo "${git_description}" | awk -F- '{print $(NF)}'  | sed 's/^g//')
+                echo "::set-output name=version::$version"
+                echo "::set-output name=revision::$revision"
+                echo "::set-output name=commit::$commit"
+                echo "::set-output name=date::$(date +"%a, %d %b %Y %T %z")"
+
+          - name: Install packaging tools
+            run: |
+                sudo apt-get update
+                sudo apt-get install software-properties-common -y
+                sudo add-apt-repository ppa:longsleep/golang-backports -y
+                sudo apt-get update
+                sudo apt-get install gnupg dput dh-make devscripts lintian golang -y
+
+          - name: Fill debian/*
+            uses: cschleiden/replace-tokens@v1
+            with:
+              files: '[".github/workflows/delivery/ubuntu/debian/*"]'
+              tokenPrefix: '{{'
+              tokenSuffix: '}}'
+            env:
+              ARCH: "any"
+              PACKAGE_NAME: "pack-cli"
+              UBUNTU_VERSION:  ${{ matrix.target }}
+              HOMEPAGE: "https://buildpacks.io"
+              PACK_VERSION: ${{ steps.metadata.outputs.version }}
+              REPO: "buildpacks/pack"
+              DESCRIPTION: "CLI for building apps using Cloud Native Buildpacks"
+              DATE_TIME: ${{ steps.metadata.outputs.date }}
+              MAINTAINER_NAME: "cncf-buildpacks"
+              MAINTAINER_EMAIL: "cncf-buildpacks-maintainers@lists.cncf.io"
+
+          - name: Create PPA
+            env:
+              PACKAGE_NAME: "pack-cli"
+              PACKAGE_VERSION: ${{ steps.metadata.outputs.version }}
+              MAINTAINER: "cncf-buildpacks"
+              MAINTANER_EMAIL: "cncf-buildpacks-maintainers@lists.cncf.io"
+              GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
+              GPG_PUBLIC_KEY: ${{ secrets.GPG_PUBLIC_KEY }}
+            run: ./.github/workflows/scripts/create_ppa.sh
+
+          - name: Test Xenial
+            if: matrix.target == 'xenial'
+            uses: docker://ubuntu:16.04
+            env:
+              PACKAGE_NAME: "pack-cli"
+            with:
+              entrypoint: .github/workflows/delivery/ubuntu/test-install-package.sh
+
+          - name: Test Bionic
+            if: matrix.target == 'bionic'
+            uses: docker://ubuntu:18.04
+            env:
+              PACKAGE_NAME: "pack-cli"
+            with:
+              entrypoint: .github/workflows/delivery/ubuntu/test-install-package.sh
+
+          - name: Test Focal
+            if: matrix.target == 'focal'
+            uses: docker://ubuntu:20.04
```

Comment:
can you use `matrix.runner` here, to simplify it if we want to change them? or is that not possible in `uses:`?

Reply:
Ya not possible to dynamically set the `uses:` fields :(

Reply:
![cry](https://media2.giphy.com/media/d2lcHJTG5Tscg/giphy.gif)